### PR TITLE
Fix incorrect typehint for Text class's color property

### PR DESCRIPTION
**Pull request by yzhong52**
## Overview: What does this pull request change?

The type should be str instead of color. If we check line 475 how it is used:

```
color = Color(color) if color else VMobject().color
```

The color class is created there.

## Reviewer Checklist
- [x] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

**Review by Darylgolden — CHANGES_REQUESTED** [review]
Thank you for your contribution! The type should be both `Color` and `str`; the user may choose to pass a `Color` class as an argument.

**Comment by yzhong52** [comment]
Not sure if I agree with that @Darylgolden. Color is being used as string everywhere I can see: 

https://docs.manim.community/en/stable/examples.html

Same as here, it is just an error in the type hint. The implementation below is also expecting `color` to be a string. 

If we do want to add support for a strong `Color` type in the public facing APIs, it should probably be a wider effort to change them all (as well as the documentation) instead of just this `Text` class here?

**Comment by yzhong52** [comment]
> That's true, such efforts have already started [as seen here](https://github.com/ManimCommunity/manim/blame/main/manim/mobject/geometry/arc.py#L483). Most of the time, these incorrect typehints are overlooked by reviewers reviewing the PRs introducing them, which explains the inconsistencies.

That sounds great! Can we merge this? I ran into issue using color for Text here and I just don't want others to run into the same bug. I am happy to close this for now too to avoid conflicts with the broader effort.

File: manim/mobject/text/text_mobject.py [hunk 1]
```diff
@@ -406,7 +406,7 @@ def __init__(
         text: str,
         fill_opacity: float = 1.0,
         stroke_width: float = 0,
-        color: Color | None = None,
+        color: str | None = None,
```
Comment (behackl) [inline manim/mobject/text/text_mobject.py:409]: The type hint should still include `Color`.

```suggestion
        color: Color | str | None = None,
```

**Review by behackl — DISMISSED** [review]

**Comment by Darylgolden** [comment]
To elaborate, something like `Color(Color(RED)) ` works perfectly fine.

**Comment by yzhong52** [comment]
> To elaborate, something like Color(Color(RED)) works perfectly fine.

Thanks for clarifying. Didn't know it works this way!

**Review by Darylgolden — APPROVED** [review]
LGTM, thank you!

**Pull request merged**